m2m/m2m.py

```diff
@@ -185,62 +181,52 @@ def writeJSONFile(self, baseDirectory, foldername, data):
 
 if __name__ == '__main__':
 
-    usage = 'usage: %prog [options] <csv file>'
+    parser = ArgumentParser()
 
-    parser = OptionParser(usage=usage)
+    parser.add_argument('mapping',
+                        help='Specify the mapping file to use for this CSV')
 
-    parser.add_option('-m', '--mapping', action='store', type='string',
-                      dest='mapping',
-                      help='Specify the mapping file to use for this CSV')
+    parser.add_argument('file',
+                        help='Specify a CSV file to process.')
 
-    parser.add_option('-n', '--row', action='store', type='int',
-                      dest='row',
-                      help='Specify a single row number to process')
+    parser.add_argument('-n', '--row', type=int,
+                        dest='row',
+                        help='Specify a single row number to process')
 
-    parser.add_option('-w', '--write', action='store_true',
-                      dest='write',
-                      help='Write records to files')
+    parser.add_argument('-w', '--write', action='store_true',
+                        dest='write',
+                        help='Write records to files')
 
-    parser.add_option('-j', '--json',
-                      action='store_true',
-                      dest='json',
-                      help='UWrite json version of metadata')
+    parser.add_argument('-j', '--json', action='store_true',
+                        dest='json',
+                        help='Write json version of metadata')
 
-    (options, args) = parser.parse_args()
-
-    if not options.mapping:
-        parser.error('You must supply a mapping file.')
-
-    if not len(args):
-        parser.error('You must supply a CSV file to process.')
-
-    print('Processing CSV file %s with mapping %s' % (args[0], options.mapping))
-
-    mappingPath = os.path.abspath(options.mapping)
-    CSVPath = os.path.abspath(args[0])
+    args = parser.parse_args()
 
+    print('Processing CSV file %s with mapping %s' % (args.file, args.mapping))
+    mappingPath = os.path.abspath(args.mapping)
+    CSVPath = os.path.abspath(args.file)
     CSVRows = CSVToDict(CSVPath)
-
     localDict = {}
 
     exec(compile(open(mappingPath).read(), mappingPath, 'exec'), {}, localDict)
 
     mappingFunction = localDict['processRecord']
 
-    if options.row:
+    if args.row:
         try:
-            CSVRows = [CSVRows[options.row - 1]]
+            CSVRows = [CSVRows[args.row - 1]]
```

Comment:
Are we intentionally allowing negative numbers?

Reply:
I don't think so. I can add a check and raise an error.